src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -462,6 +491,190 @@ private ProjectViewModel FindProject(Uri serverUri, string projectKey)
+                {
+                    return;
+                }
+
+                var solutionBinding = this.host.GetService<ISolutionBindingSerializer>();
+                solutionBinding.AssertLocalServiceIsNotNull();
+
+                var binding = solutionBinding.ReadSolutionBinding();
+                if (binding == null)
+                {
+                    return;
+                }
+
+                if (binding.Profiles == null || binding.Profiles.Count == 0)
+                {
+                    // Old binding, force refresh immediately
+                    VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheckNoProfiles);
+                    updateAction();
+                    return;
+                }
+
+                VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheck);
+
+                CancellationToken token = this.TokenSource.Token;
+                this.BackgroundTask = System.Threading.Tasks.Task.Run(() =>
+                {
+                    if (this.ProcessQualityProfileChanges(binding, languages, token))
+                    {
+                        this.host.UIDispatcher.BeginInvoke(new Action(()=>
+                        {
+                            // We mustn't update if was cancelled
+                            if(!token.IsCancellationRequested)
+                            {
+                                updateAction();
+                            }
+                        }));
+                    }
+                }, token);
+            }
+
+            private bool ProcessQualityProfileChanges(BoundSonarQubeProject binding, IEnumerable<LanguageGroup> projectLanguageGroups, CancellationToken token)
+            {
+                Debug.Assert(binding != null);
+
+                ConnectionInformation connection = binding.CreateConnectionInformation();
+                Dictionary<LanguageGroup, QualityProfile> newProfiles;
+                if (!this.TryGetNewProfiles(binding, projectLanguageGroups, token, connection, out newProfiles))
+                {
+                    VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheckFailed);
+                    return false; // Error, can't proceed
+                }
+
+                if (!newProfiles.Keys.All(binding.Profiles.ContainsKey))
+                {
+                    VsShellUtils.WriteToGeneralOutputPane(this.host, Strings.SonarLintProfileCheckSolutionRequiresMoreProfiles);
+                    return true; // Missing profile, refresh
+                }
+
+                foreach (var keyValue in binding.Profiles)
+                {
+                    LanguageGroup group = keyValue.Key;
+                    ApplicableQualityProfile oldProfileInfo = keyValue.Value;
+
+                    if (!newProfiles.ContainsKey(group))
+                    {
+                        // Not a relevant profile, we should just ignore it.
```

Comment:
No, 591 checks that all new profiles are included in the binding profile, 602 checks if the binding profile has a quality profile that was not fetched part of the new profiles since the solution has no project(s) that require that profile.